SmartDeviceLink/SDLLifecycleManager.m

```diff
@@ -364,29 +368,41 @@ - (void)didEnterStateRegistered {
         [self sdl_transitionToState:SDLLifecycleStateUnregistering];
         return;
     }
-
     NSArray<SDLLanguage> *supportedLanguages = self.configuration.lifecycleConfig.languagesSupported;
-    SDLLanguage desiredLanguage = self.configuration.lifecycleConfig.language;
-    SDLLanguage actualLanguage = self.registerResponse.language;
-    BOOL delegateCanUpdateLifecycle = [self.delegate respondsToSelector:@selector(managerShouldUpdateLifecycleToLanguage:)];
-    
+    // the language represents hmi language
+    SDLLanguage desiredHmiLanguage = self.configuration.lifecycleConfig.language;
+    SDLLanguage desiredVrLanguage = self.vrLanguage;;
+
+    SDLLanguage actualHmiLanguage = self.registerResponse.hmiDisplayLanguage;
+    SDLLanguage actualVrLanguage = self.registerResponse.language;
+
+    BOOL delegateCanUpdateLifecycle = [self.delegate respondsToSelector:@selector(managerShouldUpdateLifecycleToLanguage:hmiLanguage:)];
+    SDLLogD(@"SDLLifecycleManager sdl_checkLanguageUpdating delegateCanUpdateLifecycle: %d", delegateCanUpdateLifecycle);
     // language mismatch? but actual language is a supported language? and delegate has implemented method?
-    if (![actualLanguage isEqualToEnum:desiredLanguage] && [supportedLanguages containsObject:actualLanguage] && delegateCanUpdateLifecycle) {
-        [self sdl_transitionToState:SDLLifecycleStateUpdatingConfiguration];
-    } else {
+    if (delegateCanUpdateLifecycle && [supportedLanguages containsObject:actualHmiLanguage] && [supportedLanguages containsObject:actualVrLanguage]) {
+        if (![actualHmiLanguage isEqualToEnum:desiredHmiLanguage]
+            || ![actualVrLanguage isEqualToEnum:desiredVrLanguage]) {
+            [self sdl_transitionToState:SDLLifecycleStateUpdatingConfiguration];
+        }
+        else {
+            [self sdl_transitionToState:SDLLifecycleStateSettingUpManagers];
+        }
+    }
+    else {
         [self sdl_transitionToState:SDLLifecycleStateSettingUpManagers];
     }
 }
 
 - (void)didEnterStateUpdatingConfiguration {
     // We can expect that the delegate has implemented the update method and the actual language is a supported language
+    SDLLanguage actualHmiLanguage = self.registerResponse.hmiDisplayLanguage;
     SDLLanguage actualLanguage = self.registerResponse.language;
-    SDLLogD(@"Updating configuration due to language mismatch. New langugage: %@", actualLanguage);
-
-    SDLLifecycleConfigurationUpdate *configUpdate = [self.delegate managerShouldUpdateLifecycleToLanguage:actualLanguage];
+    SDLLogD(@"Updating configuration due to language mismatch. New langugage: %@, hmiLanguage: %@", actualLanguage, actualHmiLanguage);
 
+    SDLLifecycleConfigurationUpdate *configUpdate = [self.delegate managerShouldUpdateLifecycleToLanguage:actualLanguage hmiLanguage:actualHmiLanguage];
```

Comment:
You can't just call the new one, you have to see if they implement the new one and call that, or else check if they implement the old one and call that. Otherwise you're breaking existing apps.